cmd/src/lsif_upload.go

```diff
@@ -219,18 +219,35 @@ func (e errorWithHint) Error() string {
 	return fmt.Sprintf("%s\n\n%s\n", e.err, e.hint)
 }
 
-var errUnauthorizedHint = strings.Join([]string{
-	"You may need to specify or update your GitHub access token to use this endpoint.",
-	"See https://docs.sourcegraph.com/cli/references/lsif/upload.",
-}, "\n")
-
 // handleLSIFUploadError writes the given error to the given output. If the
 // given output object is nil then the error will be written to standard out.
 //
 // This method returns the error that should be passed back up to the runner.
 func handleLSIFUploadError(out *output.Output, err error) error {
 	if err == upload.ErrUnauthorized {
-		err = errorWithHint{err: err, hint: errUnauthorizedHint}
+		var actionableHints []string
+		if lsifUploadFlags.gitHubToken != "" {
+			actionableHints = append(actionableHints,
+				"The supplied -github-token did not give the expected access level for the target repository.",
+				"Please check that supplied token matches the code host and has expected access of the target repository.",
```

Comment:
"expected access level" seems quite vague/uninformative. 😐 Any specific reason to not repeat the exact requirements here? (I can see one reason to not duplicate the docs; that would make keeping them in sync trickier.)

Reply:
The scopes are different for github+gitlab. Should we just fork on each one and have explicit instructions here?

Reply:
That sounds like a good approach. 👍